src/main/java/com/hierynomus/smbj/share/File.java

```diff
@@ -148,6 +150,20 @@ public void read(OutputStream destStream, ProgressListener progressListener) thr
         is.close();
     }
 
+    /***
+     * The function for truncate or set file length for a file
+     * @param endOfFile 64-bit signed integer in bytes, MUST be greater than or equal to 0
+     * @throws SMBApiException
+     */
+    public void setLength(long endOfFile) throws SMBApiException {
+        try {
+            FileEndOfFileInformation endOfFileInfo = new FileEndOfFileInformation(endOfFile);
+            this.setFileInformation(endOfFileInfo);
+        } catch (IllegalArgumentException iae) {
```

Comment:
Sorry, I didn't got what you mean clearly. Do you mean I shouldn't catch the exception and just let it throws the exception to the caller (let the caller to handle the exception by itself)?